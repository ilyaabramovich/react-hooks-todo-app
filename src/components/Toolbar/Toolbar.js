import React from "react";
import "./Toolbar.css";

export const Toolbar = ({ count, onClear }) => {
  return (
    <div className="toolbar page__toolbar">
      <span>{`${count} item${count > 1 ? "s" : ""} left`}</span>
      <button className="button toolbar__button" onClick={onClear}>
        Clear all
      </button>
    </div>
  );
};
