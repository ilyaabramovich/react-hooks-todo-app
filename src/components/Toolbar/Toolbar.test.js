import React from "react";
import { render } from "@testing-library/react";
import { Toolbar } from "./Toolbar";

test("it renders", () => {
  const { asFragment } = render(<Toolbar count={1} />);
  expect(asFragment()).toMatchSnapshot();
});

test("displays items count correctly", () => {
  const { getByLabelText } = render(<Toolbar count={1} />);
  expect(getByLabelText("TODO counter")).toHaveTextContent("1 item left");
});
