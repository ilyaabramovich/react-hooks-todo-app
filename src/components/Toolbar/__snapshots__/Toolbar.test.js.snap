// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<DocumentFragment>
  <div
    class="toolbar page__toolbar"
  >
    <span
      aria-label="TODO counter"
    >
      1 item left
    </span>
    <button
      class="button toolbar__button"
    >
      Clear all
    </button>
  </div>
</DocumentFragment>
`;
